### PR TITLE
Refine ProbeHelper for CloudAuditLogsSource probe

### PR DESCRIPTION
**Pull request by tommyreddad**
Refine ProbeHelper for latest design of CloudAuditLogsSource probe

## Proposed Changes

- ProbeHelper only creates pub/sub topics instead of also deleting them.
- Add tests for unrecognized probe event subjects to avoid channel leaking in case of malformed events.

### CODE REVIEW

File: test/test_images/probe_helper/probe_helper_test.go [hunk 1]
```diff
@@ -420,6 +409,14 @@ func TestProbeHelper(t *testing.T) {
 				wantResult: cloudevents.ResultACK,
 			},
 		},
+	}, {
+		name: "Unrecognized Broker E2E delivery probe subject",
+		steps: []eventAndResult{
+			{
+				event:      probeEvent("broker-e2e-delivery-probe", "invalid-subject"),
```
Comment (liu-cong) [inline test/test_images/probe_helper/probe_helper_test.go:416]: nice! We can continue to improve test coverage for error cases

File: test/test_images/probe_helper/probe_helper.go [hunk 2]
```diff
@@ -249,17 +258,9 @@ func (ph *ProbeHelper) forwardFromProbe(ctx context.Context) cloudEventsFunc {
...
-				if _, err := ph.pubsubClient.CreateTopic(ctx, event.ID()); err != nil {
-					return logNACK(ctx, "Probe forwarding failed, error creating pubsub topic %v: %v", event.ID(), err)
-				}
-			case CloudAuditLogsSourceProbeDeleteSubject:
-				// Delete the pubsub topic with the given ID.
-				if err := ph.pubsubClient.Topic(event.ID()).Delete(ctx); err != nil {
-					return logNACK(ctx, "Probe forwarding failed, error deleting pubsub topic %v: %v", event.ID(), err)
-				}
+			// Create a pubsub topic with the given ID.
+			if _, err := ph.pubsubClient.CreateTopic(ctx, event.ID()); err != nil {
```
Comment (liu-cong) [inline test/test_images/probe_helper/probe_helper.go:262]: I think the topic name should be "cloudauditlogssource-probe-"+event.ID()?

Comment (tommyreddad) [reply test/test_images/probe_helper/probe_helper.go:262]: The event ID already includes the probe type, so event.ID() looks something like "cloudauditlogssource-probe-1234567890"

**Pull request merged**